main/src/cgeo/geocaching/Image.java

```diff
@@ -15,27 +17,92 @@
 
 import java.io.File;
 
+/**
+ * Represent an Image along with Title and Description.
+ * Mostly used for representing image in/from logs.
+ */
 public class Image implements Parcelable {
-    private final String url;
-    private final String title;
-    private final String description;
+    @NonNull private Uri uri;
```

Comment:
For references http://programmers.stackexchange.com/questions/151733/if-immutable-objects-are-good-why-do-people-keep-creating-mutable-objects

If I understand well the concept, I should create new objects with previous data, and only change the necessary fields in constructor?

Is some sort of function in `Image` class called `ImageBuilder(???)` (or whatever) is the way to go? maybe that's what you say here "we should use mutable builders to build immutable data structures"